packages/babel-parser/src/types.js

```diff
@@ -1040,6 +1042,8 @@ export type TsTypeParameter = NodeBase & {
   type: "TSTypeParameter",
   // TODO(Babel-8): remove string type support
   name: string | Identifier,
+  in?: boolean,
+  out?: boolean,
```

Comment:
I'm not 100% sold on using boolean flags as it locks us in with naming but I guess that this is the same thing we did with class property modifiers as well so it's not like we're breaking convention in the AST.

Given the precedence - this is probably the best approach, so LGTM